src/plugins/explore/public/components/query_panel/actions/ppl_execute_query_action.tsx

```diff
@@ -54,15 +136,58 @@ export function usePPLExecuteQueryAction(
         const shouldExecute = args.autoExecute !== false; // Default to true
 
         if (shouldExecute) {
+          // Get the current query state to determine the cache key
+          const state = services.store.getState();
+          const query = state.query;
+
+          // Prepare the query object to get the cache key
+          const queryObject = {
+            ...query,
+            query: args.query,
+          };
+          const cacheKey = defaultPrepareQueryString(queryObject);
```

Comment:
Shall we use `args.query` as cache key? `defaultPrepareQueryString(queryObject)` is not the same as the execution query in the explore page.